TransferMethod/Sources/View/Widget/ExpiryDateWidget.swift

```diff
@@ -20,43 +20,54 @@ import UIKit
 final class ExpiryDateWidget: TextWidget {
-    private var pickerView: ExpiryDatePickerView!
-    private let toolbar = UIToolbar()
+    private static let dateTextFieldFormat = "##/##"
+    private static let dateApiFormat = "yyyy-MM"
+    private static let placeholderText = "MM/YY"
+    private static var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MM/yy"
+        return formatter
+    }()
 
     override func setupLayout(field: HyperwalletField) {
         super.setupLayout(field: field)
-        setupPickerView(field: field)
-        toolbar.setupToolBar(target: self, action: #selector(self.doneButtonTapped))
         setupTextField()
     }
 
-    /// format the expiryDate to YYYY-MM, which can be accepted by server through REST API call
-    ///
-    /// - Returns: formatted expiryDate
-    override func value() -> String {
-        if textField.text?.isEmpty ?? true {
-            return ""
+    override func textFieldDidChange() {
+        let text = getUnformattedText()
+        if !text.isEmpty {
+            textField.text = formatDisplayString(with: ExpiryDateWidget.dateTextFieldFormat, inputText: text)
         } else {
-            return String(format: "%04d-%02d", pickerView.year, pickerView.month)
+            textField.text = ""
+            textField.placeholder = ExpiryDateWidget.placeholderText
         }
     }
 
-    @objc
-    private func doneButtonTapped() {
-        guard let month = pickerView.month, let year = pickerView.year else {
-            return
+    override func textFieldDidEndEditing(_ textField: UITextField) {
+        super.textFieldDidEndEditing(textField)
+        if textField.text?.isEmpty ?? true {
+            textField.placeholder = ExpiryDateWidget.placeholderText
+        }
+    }
+
+    /// format the expiryDate to YYYY-MM, which can be accepted by server through REST API call
+    ///
+    /// - Returns: formatted expiryDate
+    override func value() -> String {
+        if let text = textField.text, !(textField.text?.isEmpty ?? true) {
+            return formatExpiryDateForApi(text)
         }
-        textField.text = String(format: "expiry_date_format".localized(), month, year % 100)
-        textField.resignFirstResponder()
+        return ""
     }
 
-    private func setupPickerView(field: HyperwalletField) {
-        pickerView = ExpiryDatePickerView(value: field.value)
-        pickerView.accessibilityIdentifier = "expiryDateWidgetPicker"
+    func formatExpiryDateForApi(_ text: String) -> String {
+        let date = ExpiryDateWidget.dateFormatter.date(from: text)
+        return date?.formatDateToString(dateFormat: ExpiryDateWidget.dateApiFormat) ?? ""
     }
 
     private func setupTextField() {
-        textField.inputView = pickerView
-        textField.inputAccessoryView = toolbar
+        textField.placeholder = ExpiryDateWidget.placeholderText
+        textField.keyboardType = UIKeyboardType.numberPad
```

Comment:
Suggestion: You could have simple extended from NumberWidget instead of setting it here.

Reply:
oh right.. done